File: Data Structures & Algorithms/max-water-container/submission-1.js
class Solution {
    /**
     * @param {number[]} heights
     * @return {number}
     */
    maxArea(heights) {
        let n = heights.length
        let left = 0;
        let right = n-1;
        let maxVal =0
        while(left < right){
            let currentArea = (Math.min(heights[left], heights[right])) * (right-left);
            maxVal = Math.max(currentArea,maxVal)
            heights[left] > heights[right] ? right-- : left++ 
        }
        return maxVal
    }
}

File: Data Structures & Algorithms/buy-and-sell-crypto/submission-1.java
class Solution {
    public int maxProfit(int[] prices) {
        int l = 0;
        int maxProfit = 0;

        for(int r =1;r<prices.length;r++ ){
            if(prices[r] > prices[l]){
                //find max
                maxProfit =  Math.max(maxProfit, prices[r]-prices[l]);
            }else{
                l = r;
            }

        }
        return maxProfit;
    }
}
